app/utils/handleFileUpload.ts

```diff
@@ -0,0 +1,20 @@
+import { EvolverConfigWithoutDefaults } from "client";
```

Comment:
`handleFileUpload` sounds maybe too generic. Might change to `handleEvolverConfigFileUpload` or the like

Reply:
i made this more generic, ht "evolver config" is now a generic type on the fileUpload helper.

Reply:
```typescript
import React from "react";

export function handleFileUpload<T>({
  e,
  setData,
}: {
  e: React.ChangeEvent<HTMLInputElement>;
  setData: React.Dispatch<React.SetStateAction<T>>;
}) {
  const fileReader = new FileReader();
  if (e.target.files?.[0]) {
    fileReader.readAsText(e.target.files[0], "UTF-8");
    fileReader.onload = (event) => {
      const uploadedConfig = JSON.parse(event.target?.result as string);
      setData(uploadedConfig);
    };
  }
}
```